store/store.go

```diff
@@ -171,6 +183,74 @@ func (bs *BlockStore) LoadSeenCommit(height int64) *types.Commit {
 	return commit
 }
 
+// PruneBlocks removes block up to (but not including) a height. It returns number of blocks pruned.
+func (bs *BlockStore) PruneBlocks(height int64) (uint64, error) {
+	bs.mtx.RLock()
+	if height > bs.height {
+		bs.mtx.RUnlock()
+		return 0, fmt.Errorf("cannot prune the latest height %v", bs.height)
+	}
+	base := bs.base
+	bs.mtx.RUnlock()
+	if base == 0 {
+		base = 1
+	}
+	if base > height {
```

Comment:
that shouldn't even be possible, no? if so, maybe change it to assert (panic)

Reply:
It's possible if the block store is empty, where in this case base=1 and height=0.

Reply:
That's up to the caller. If the block store is at base 100 and the caller tries to prune to height 50, I think that should be an error. It generally shouldn't happen, but there could be e.g. race conditions. We should avoid panics in such situations, since it can turn an otherwise minor bug into a crash bug.